test/runtests.jl

```diff
@@ -1,200 +1,32 @@
-@unix_only byte_file_contents = string("466F6F426172", "0D0A", "466F6F426172")
-@unix_only byte_file_contents_ascii_transfer = string("466F6F426172", "0A", "466F6F426172")
-@windows_only byte_file_contents = string("466F6F426172", "0A", "466F6F426172", "1A1A1A")
-byte_file_name = randstring(20)
-byte_upload_file_name = randstring(21)
-upload_file_name = "test_upload.txt"
-upload_file_contents = randstring(100)
-open(upload_file_name, "w") do f
-    write(f, upload_file_contents)
-end
-
-@unix_only upload_local_byte_file_contents = string("466F6F426172", "0A", "466F6F426172")
-@unix_only upload_local_byte_file_contents_ascii_transfer = string("466F6F426172", "0D0A", "466F6F426172")
-@windows_only upload_local_byte_file_contents = string("466F6F426172", "0D0A", "466F6F426172")
-@windows_only upload_local_byte_file_contents_ascii_transfer = string("466F6F426172", "0A", "466F6F426172")
-
-if (length(ARGS) >= 1 && ARGS[1] == "true")
-    test_ssl = true
-else
-    test_ssl = false
-end
-
-if (length(ARGS) >= 2 && ARGS[2] == "true")
-    test_implicit = true
-else
-    test_implicit = false
-end
-
-if (length(ARGS) == 4)
-    user = ARGS[3]
-    pswd = ARGS[4]
-end
-
-if (test_ssl)
-    if (test_implicit)
-        fp = joinpath(dirname(@__FILE__), "test_implicit_ssl.jl")
-        println("$fp ...\n")
-        include(fp)
-    else
-        fp = joinpath(dirname(@__FILE__), "test_explicit_ssl.jl")
-        println("$fp ...\n")
-        include(fp)
-    end
-else
-    # Start Java, and point to the class in this directory
-    pkg_dir = joinpath(dirname(@__FILE__), "..")
-    JavaCall.init([
-        "-Djava.class.path=$(joinpath(pkg_dir, "test"))",
-        "-Djava.ext.dirs=$(joinpath(pkg_dir, "deps", "ext"))",
-    ])
-    MockFTPServerJulia = @jimport MockFTPServerJulia
-
-    set_user(user, pswd, home_dir)
-    set_files()
-    set_command_response("AUTH", 230, "Login successful.")
-    port = start_server()
-    host = "$original_host:$port"
+using Base.Test
 
-    # Note: If LibCURL complains that the server doesn't listen it probably means that
-    # the MockFtpServer isn't ready to accept connections yet.
 
-    @testset "FTPCLient" begin
+include("server/server.jl")
+include("utils.jl")
 
-        include("test_non_ssl.jl")
-        include("test_ftp_object.jl")
+setup_server()
+ftp_init()
+server = FTPServer()
 
-        # Basic commands will now error
-        set_errors()
+testdir = "test_dir"
+upload_file = "test_upload.txt"
+download_file = "test_download.txt"
+tempfile(upload_file)
+tempfile(joinpath(ROOT,download_file))
 
-        include("test_client_errors.jl")
+cleanup_file(download_file)
+cleanup_file(joinpath(ROOT, upload_file))
 
-    end
+include("ssl.jl")
+include("non_ssl.jl")
+include("ftp_object.jl")
 
-    stop_server()
-    JavaCall.destroy()
+cleanup_file(upload_file)
+cleanup_file(joinpath(ROOT, download_file))
 
-end
+ftp_cleanup()
+teardown_server()
 
-# Done testing
-rm(upload_file_name)
+close(server)
```

Comment:
If a test fails then the server won't be shutdown. You should wrap most of this in a try/finally:
```julia
server = FTPServer()
try
    ...
finally
    close(server)
end
```